petdeface/petdeface.py

```diff
@@ -1,31 +1,31 @@
 import argparse
-import shutil
-import os
-import sys
 import json
-import subprocess
+import os
 import pathlib
-import bids
-import gzip
+import shutil
+import subprocess
+import sys
 from typing import Union
-from nipype import Function
-from nipype.interfaces.io import SelectFiles
-
-# some day someone will figure out how to make packing work across dev and install environments
-try:
-    from mideface import Mideface
-    from pet import create_weighted_average_pet
-except ModuleNotFoundError:
-    from .mideface import Mideface
-    from .pet import create_weighted_average_pet
```

Comment:
I know this stuff is goofy, but removing it breaks this when running it in development mode, see:

```(petdeface-py3.10) petdeface % poetry install --sync
Configuration file exists at /Library/Preferences/pypoetry, reusing this directory.

Consider moving TOML configuration files to /Library/Application Support/pypoetry, as support for the legacy directory will be removed in an upcoming release.
Installing dependencies from lock file

Package operations: 17 installs, 4 updates, 1 removal

  • Removing bids (0.0)
  • Updating numpy (1.24.4 -> 1.25.2)
  • Updating setuptools (67.8.0 -> 68.1.2)
  • Installing distlib (0.3.7)
  • Installing lazy-loader (0.3)
  • Installing platformdirs (3.10.0)
  • Updating tifffile (2023.7.10 -> 2023.8.25)
  • Installing cfgv (3.4.0)
  • Installing identify (2.5.27)
  • Installing mccabe (0.7.0)
  • Installing mypy-extensions (1.0.0)
  • Installing nodeenv (1.8.0)
  • Installing pathspec (0.11.2)
  • Installing pycodestyle (2.11.0)
  • Installing pyflakes (3.1.0)
  • Updating scikit-image (0.19.3 -> 0.21.0)
  • Installing tomli (2.0.1)
  • Installing virtualenv (20.24.3)
  • Installing black (23.7.0)
  • Installing flake8 (6.1.0)
  • Installing isort (5.12.0)
  • Installing pre-commit (3.3.3)

Installing the current project: petdeface (0.0.0)
(petdeface-py3.10) petdeface % python petdeface/petdeface.py 
Traceback (most recent call last):
  File "/Projects/petdeface/petdeface/petdeface.py", line 20, in <module>
    from .mideface import Mideface
ImportError: attempted relative import with no known parent package```

Reply:
Hi @bendhouseart, running `python petdeface/petdeface.py` does give me the same error. Instead of this, would using the petdeface cli within the virtualenv set up by poetry be an acceptable alternative? i.e., `poetry run petdeface -h`. Since petdeface module is installed within this virtualenv, there is no `ImportError`. Otherwise, I can revert this edit.

Reply:
It's solely for debugging and while a try catch doesn't seem all that elegant I've found that it works pretty well. Perhaps I shouldn't misinterpret a failure of my footgun to fire as it being safe, but I've never really bothered to find a better way to handle imports that make packaging and using an ide's debugger work harmoniously.

Let's kick the can and do proper imports in a smaller atomic PR after merging this one.